zeroconf/src/macos/browser.rs

```diff
@@ -203,11 +206,14 @@ unsafe fn handle_resolve(
         None
     };
 
+    println!("host_target: {}", c_str::copy_raw(host_target));
+    println!("interface_index: {}", interface_index);
+
     ManagedDNSServiceRef::default().get_address_info(
         GetAddressInfoParams::builder()
             .flags(bonjour_sys::kDNSServiceFlagsForceMulticast)
             .interface_index(interface_index)
-            .protocol(0)
+            .protocol(1)
```

Comment:
From apple docs:
>Pass in kDNSServiceProtocol_IPv4 to look up IPv4 addresses, or kDNSServiceProtocol_IPv6 to look up IPv6 addresses, or both to look up both kinds. If neither flag is set, the system will apply an intelligent heuristic, which is (currently) that it will attempt to look up both, except:

However, I've also experienced that this "intelligent heuristic" is not always so intelligent so I think I understand your motivation for your change. My preference, however, would be to expose this to the API instead of forcing IPv4. Such an API should likely be an enum that is a member-field of `BonjourMdnsBrowser` and a getter/setter in that same impl (unless Avahi has an analogous system in which case we would put the getter/setter in the trait similar to `NetworkInterface`.

If you would like to implement this, awesome! If not, I'm happy to allow this change and save that for a later date.

Reply:
I reverted it, because I'm not sure whats needs to be done for the implementation you mentioned 🙈. And I also think it should not be mixed up with this PR.

Reply:
Are you sure? This might start giving you the problem where you get IPv4 sometimes and IPv6 other times. I've experienced this myself. Even if the abstraction is not implemented here, I still think it's a good idea to make IPv4 (1) the default.

Reply:
Okay, then we enforce IPv4